_cffi_build/build.py

```diff
@@ -36,14 +34,13 @@ def _mk_ffi(sources, name="_libevmjit", bundled=True, **kwargs):
 _base = [Source(absolute("evm.h"), "#include \"_cffi_build/evm.h\"",)]
 ffi = _mk_ffi(_base, libraries=['evmjit'])
 ffi.cdef("""
-    extern "Python" void  evm_query(struct evm_env* env,
+    extern "Python" union evm_variant evm_query(struct evm_env* env,
                                           enum evm_query_key key,
-                                          union evm_variant* arg,
-                                          union evm_variant* ret);
+                                          union evm_variant arg);
```

Comment:
CFFI confirmed that your case should work: https://bitbucket.org/cffi/cffi/issues/281/clarify-docs-using-unions-in-extern-python